craft_archives/repo/package_repository.py

```diff
@@ -70,22 +65,22 @@ def _alias_generator(value: str) -> str:
     return value.replace("_", "-")
 
 
-class PackageRepository(pydantic.BaseModel, abc.ABC):
+class PackageRepository(BaseModel, abc.ABC):
     """The base class for package repositories."""
 
-    class Config:  # pylint: disable=too-few-public-methods
-        """Pydantic model configuration."""
-
-        validate_assignment = True
-        allow_mutation = False
-        allow_population_by_field_name = True
-        alias_generator = _alias_generator
-        extra = "forbid"
+    model_config = ConfigDict(
+        validate_assignment=True,
+        frozen=True,
+        populate_by_name=True,
+        alias_generator=_alias_generator,
+        extra="forbid",
+    )
 
     type: Literal["apt"]
-    priority: Optional[PriorityValue]
+    priority: Optional[PriorityValue] = None
 
-    @root_validator
+    @model_validator(mode="before")
```

Comment:
I believe this can be simplified to an `after` validator, which is probably preferable:

```
    @model_validator(mode="after")
    def priority_cannot_be_zero(self, info: ValidationInfo) -> Self:
        """Priority cannot be zero per apt Preferences specification."""
        if self.priority == 0:
            raise _create_validation_error(
                url=str(info.data.get("url") or info.data.get("ppa") or info.data.get("cloud")),
                message="invalid priority: Priority cannot be zero.",
            )
        return self
```